e2e/aks_model.go

```diff
@@ -201,7 +210,101 @@ func privateEndpointExists(ctx context.Context, t *testing.T, nodeResourceGroup,
 	return false, nil
 }
 
-func createPrivateEndpoint(ctx context.Context, t *testing.T, nodeResourceGroup, privateEndpointName string, vnet VNet) (armnetwork.PrivateEndpointsClientCreateOrUpdateResponse, error) {
+func createPrivateAzureContainerRegistry(ctx context.Context, t *testing.T, nodeResourceGroup, privateACRName string) error {
+	t.Logf("Creating private Azure Container Registry in rg %s\n", nodeResourceGroup)
+
+	createParams := armcontainerregistry.Registry{
+		Location: to.Ptr(config.Config.Location),
+		SKU: &armcontainerregistry.SKU{
+			Name: to.Ptr(armcontainerregistry.SKUNamePremium),
+		},
+		Properties: &armcontainerregistry.RegistryProperties{
+			AdminUserEnabled:     to.Ptr(false),
+			AnonymousPullEnabled: to.Ptr(true), // required to pull images from the private ACR without authentication
+		},
+	}
+
+	_, err := config.Azure.RegistriesClient.Get(ctx, nodeResourceGroup, privateACRName, nil)
+	if err != nil && strings.Contains(err.Error(), "ResourceNotFound") {
```

Comment:
`strings.Contains` is generally not a great way to check error.

The error most likely is a custom type with a field containing error code. I usually put a breakpoint and run debugger to quickly check what I can use from it. Then you can use `errors.As` and check error code (or dig into documentation/source code)